api/v1alpha1/gateway_types.go

```diff
@@ -146,14 +146,15 @@ const (
 // - aws: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html#describe-ssl-policies
 // - azure: https://docs.microsoft.com/en-us/azure/app-service/configure-ssl-bindings#enforce-tls-1112
 type ListenerTLS struct {
-	// Certificates is a list of certificates containing resources
-	// that are bound to the listener.
+	// Certificate is a reference to a Kubernetes object containing the identity
+	// certificate bound to the listener. Certificate is used when an HTTPRoute
+	// does not specify a certificate.
 	//
-	// If apiGroup and kind are empty, will default to Kubernetes Secrets resources.
+	// If apiGroup and kind are empty, defaults to a Kubernetes Secret resource.
 	//
 	// Support: Core (Kubernetes Secrets)
 	// Support: Implementation-specific (Other resource types)
-	Certificates []core.TypedLocalObjectReference `json:"certificates,omitempty"`
+	Certificate core.TypedLocalObjectReference `json:"certificate,omitempty"`
```

Comment:
We should specify if this is required or optional.
I think this should be required.

This is essentially the fallback certificate for any request and has the lowest preference.

There is another issue though. It should be possible to specify SNI <-> secret mapping at this level as well to specify different certificates for different domains.

Reply:
> I think this should be required.

@hbagdi TLS is optional for a listener. For example, a listener is configured to listen on port 80 for HTTP. Maybe `GatewaySpec` should have a `Certificates` or `DefaultCertificates` field for a gateway to use as "fall back" certs?

Reply:
My bad for not explaining properly.
What I mean is that if it is a TLS listener, then we should encourage user to provide a default certificate, which takes the lowest priority. As you said, it's the "fall back" cert.

Having more than one default certificate means that you need to have some sort of `SNI` mapping to those certs. Both cases make sense to me.